pkg/machine/qemu.go

```diff
@@ -90,11 +90,9 @@ func (q *QEMU) Create(ctx context.Context) error {
 
 	q.process = qemu
 
-	monitorCtx, cancelFunc := context.WithCancel(ctx)
-	defer cancelFunc()
-	monitor(monitorCtx, qemu, q.machineConfig.OnFailure)
```

Comment:
This handling was wrong. The `Run` call below is asynchronous (doesn't block) so the defer was always called too early, before the VM was stopped, thus the monitoring stopped early and never got to run the fail function.